searchcore/src/vespa/searchcore/proton/reference/gid_to_lid_change_listener.cpp

```diff
@@ -25,23 +25,23 @@ GidToLidChangeListener::~GidToLidChangeListener()
 }
 
 void
-GidToLidChangeListener::notifyPutDone(document::GlobalId gid, uint32_t lid)
+GidToLidChangeListener::notifyPutDone(Context context, document::GlobalId gid, uint32_t lid)
 {
-    std::promise<void> promise;
-    auto future = promise.get_future();
     _attributeFieldWriter.executeLambda(_executorId,
-                                        [this, &promise, gid, lid]() { _attr->notifyReferencedPut(gid, lid); promise.set_value(); });
-    future.wait();
+                                        [this, context=std::move(context), gid, lid]() {
```

Comment:
There might be problems if the listener can be destroyed before the task has been completed. Consider extending the destructor to sync the executor (_attributeFieldWriter) in order to ensure that all previously queued tasks have a valid listener. An alternate approach might be capture a copy of `_attr` in the lambda.

Reply:
Fixed by sync in destructor.